docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+
+## 4.0 Implementation
+
+The following section will go into further details on how the different features of **LifEasier** were implemented.
+
+### 4.1 Adding Lessons, Events, Deadlines (Fairuz)
+
+The `addLesson`, `addEvent` and `addDeadline` commands adds the specific task into the `TaskList`.
+
+##### Implementation
+
+Due to the different parameters required by the addition of different types of tasks, each command has
+ a specific intended parameter to be input by the user.
+
+For example, if the user uses the `addEvent` command, the user will have to input the `EVENT_NAME`,
+ `START_TIME` and `END_TIME`, whereas if the user uses the `addDeadline` command, the user will have input `DEADLINE_NAME` and `BY`.
+
+When the command is called, the class invokes the `TaskList` class method to add the specific task to
+ the `TaskList`, after which `Storage` is called to save the updated `TaskList` to the local save file. Figure 4.1-1 shows the 
+ sequence diagram for the `addDeadlineCommand`. This sequence diagram is applicable to the other two `addEventCommand` and 
+ `addLessonCommand`, with the only difference being in their parameters.
+ 
+ ![Figure 4.1-1](images/DeveloperGuide/Figure%204.1-1.png)  
+ _Figure 4.1-1: Sequence diagram for addDeadlineCommand execution_
+
+##### Design Considerations
+
+The success and accuracy of the command is heavily dependent on the values passed to the command by the `Parser` class.
+ Thus, checks have to be thoroughly conducted by the `Parser` class before the `Command` class is invoked.
+
+### 4.2 Editing Lessons, Events, Deadlines (Fairuz)
+
+The `editEventCommand`, `editDeadlineCommand` and `editLessonCommand` allows the user to edit existing
+ tasks in the `TaskList` according to the type of task.
+
+##### Implementation
+
+These command classes will be invoked when the user inputs the `editTask` command, followed by
+ the type of task to be edited. 
+
+The user also has an option to input the task’s name, or part of it, after which **LifEasier**
+ will print out all tasks matching the type, and the user query. If the user chooses to leave
+ that search parameter blank, **LifEasier** will print out all the tasks of the intended type.
+ The program will then prompt User to input the index of the task to be edited.
+
+Upon receiving a valid index of a task to be edited, **LifEasier** will prompt the User for the
+ parameter to be edited - `Name` or `Time`. 
+
+Upon receiving a valid option of parameter to edit, **LifEasier** will prompt the User to input
+ the new descriptions. If the User input is valid according to the required format, **LifEasier** will print a confirmation
+ of the edit. The edited task will then be saved by `Storage`. Figure 4.2-1 illustrates the flow of `editDeadlineCommand` 
+ through a sequence diagram. The logic of this class remains the same among the different types of `Tasks`.
+
+![Figure 4.2-1](images/DeveloperGuide/Figure%204.2-1.png)    
```

Comment:
![image](https://user-images.githubusercontent.com/60388723/97528974-ebdaee00-19e9-11eb-8237-a799cfca7899.png)
Would the selfInvocted method return something to the main activation bar?